zetaclient/chains/evm/observer/v2_inbound.go

```diff
@@ -160,6 +160,12 @@ func (ob *Observer) checkEventProcessability(event *gatewayevm.GatewayEVMDeposit
 
 // newDepositInboundVote creates a MsgVoteInbound message for a Deposit event
 func (ob *Observer) newDepositInboundVote(event *gatewayevm.GatewayEVMDeposit) types.MsgVoteInbound {
+	// if event.Asset is zero, it's a native token
+	coinType := coin.CoinType_ERC20
+	if event.Asset == (ethcommon.Address{}) || event.Asset.Hex() == constant.EVMZeroAddress {
```

Comment:
The two of `event.Asset == (ethcommon.Address{})` and `event.Asset.Hex() == constant.EVMZeroAddress` are equal under the current implementation of `ethcommon.Address`, but it might change in future.

If we want a unified logic across code base, maybe we can create a func `IsEmptyAddress(address ethcommon.Address)` in `pkg` to do it?